src/aria2p/api.py

```diff
@@ -850,3 +852,52 @@ def stop_listening(self) -> None:
         if self.listener:
             self.listener.join()
         self.listener = None
+
+    def input_downloads(self, lines):
+        """
+        Helper to split downloads in an input file.
+
+        Arguments:
+             lines:  URI string.
+        """
+        block = []
+        for line in lines:
+            if line.lstrip().startswith("#"):  # Ignore Comments
+                continue
+            if not line.strip():  # Ignore empty line
+                continue
+            if not line.startswith(" "):  # Read download option
+                if block:
+                    yield block
+                    block = []
+            block.append(line.rstrip("\n"))
+        if block:
+            yield block
+
+    def add_input_file(self, input_file):
+        """
+        Parse file with uris or aria2c input file.
+
+        Arguments:
+            input_file: Path to file with URIs or aria2c input file.
+
+        Returns:
+            List of tuples containing list of uris and dictionary with aria2c options.
+
+        """
+
+        downloads = []
+        with Path(input_file).open() as fd:
+            for download_lines in self.input_downloads(fd):
+                uris = download_lines[0].split("\t")
+                options = {}
+                try:
+                    for opt_line in download_lines[1:]:
+                        opt_name, opt_value = opt_line.split("=")
+                        options[opt_name.lstrip()] = opt_value
+                    downloads.append((uris, options))
+                except ValueError as error:
+                    logger.exception(error)
+                    # Notify the user of an error.
```

Comment:
This is not really useful to the user.
I suggest something like this instead:

```python
logger.error(f"Skipping download because of invalid option line '{opt_line}'")
logger.opt(exception=True).trace(error)
```

Not sure `opt_line` would be available in this scope though. This will maybe need a small refactor.

You can also remove the comment.